search.go

```diff
@@ -41,11 +41,11 @@ type QueryPart interface {
 	Name() string
 
 	// Seek returns the key for cursor.Seek
-	Seek() (Key, error)
+	Seek() Scalar
```

Comment:
why does it return a scalar (or anything at all)? Seek is usually an operation (at least when thinking of spinning/sequenced storage) to move a cursor to a set position?

Reply:
it returns the seek value where the cursor needs to move to.